### PR TITLE
KAZOO-4193 Participant events for web-sockets

### PR DESCRIPTION
**Pull request by jamhed**
* Propagate participant events to websocket
* Bug fix (handling participant sync)

### CODE REVIEW

File: applications/conference/src/conf_participant.erl [hunk 1]
```diff
@@ -449,79 +449,41 @@ code_change(_OldVsn, Participant, _Extra) ->
 %%% Internal functions
 %%%===================================================================
 
+join_conference('true'=_Moderator, ParticipantNo, Conference) ->
```
Comment (lazedo) [inline applications/conference/src/conf_participant.erl]: spec and maybe change `ParticipantNo` to `ParticipantId`

File: applications/conference/src/conf_participant.erl [hunk 2]
```diff
@@ -449,79 +449,41 @@ code_change(_OldVsn, Participant, _Extra) ->
...
 sync_participant(JObj, Call, #participant{in_conference='false'
                                           ,conference=Conference
+                                          ,discovery_event=DiscoveryEvent
                                          }=Participant) ->
-    Participator = kz_json:get_value(<<"Participant">>, JObj),
-    IsModerator = kapps_conference:moderator(Conference),
-    case Participator of
-        Moderator when IsModerator ->
-            Focus = kz_json:get_value(<<"Focus">>, JObj),
-            C = kapps_conference:set_focus(Focus, Conference),
```
Comment (lazedo) [inline applications/conference/src/conf_participant.erl]: i don't see `set_focus` in the changed code, are we dropping this ?

Comment (jamhed) [inline applications/conference/src/conf_participant.erl]: Yeah, somehow I lost it while struggling with commits. It definitely was there.

File: applications/conference/src/conf_participant.erl [hunk 3]
```diff
@@ -449,78 +449,43 @@ code_change(_OldVsn, Participant, _Extra) ->
...
-
--spec sync_member(kz_json:object(), kapps_call:call(), participant()) -> participant().
-sync_member(JObj, Call, #participant{conference=Conference
-                                     ,discovery_event=DiscoveryEvent
-                                    }=Participant) ->
-    ParticipantId = kz_json:get_value(<<"Participant-ID">>, JObj),
-    lager:debug("caller has joined the local conference as member ~p", [ParticipantId]),
-    Deaf = not kz_json:is_true(<<"Hear">>, JObj),
-    Muted = not kz_json:is_true(<<"Speak">>, JObj),
-    gen_listener:cast(self(), 'play_announce'),
```
Comment (lazedo) [inline applications/conference/src/conf_participant.erl]: `gen_listener:cast(self(), 'play_announce')`is missing from new implementation

**Pull request closed without merging**